### ISSUE TITLE
Add dueling DQN

### ISSUE THREAD

**Issue by dnabanita7** [JuliaReinforcementLearning/ReinforcementLearningZoo.jl#73]
The algorithm for dueling DQN is same as DQN except for when it is split up after the fully connected layers in the ``model()``. So, I have made changes in the src/experiments/atari.jl and src/experiments/rl_envs.jl

**Comment by findmyway** [comment]
I agree. Besides, you can also add a `DuelingNet` similar to [ImplicitQuantileNet](https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/74c6e5a0933b8168f1a60f8ce95641d24491723a/src/algorithms/dqns/iqn.jl#L10-L28) into `dqn.jl`.

**Comment by dnabanita7** [comment]
I am finding it difficult to understand the architecture in Julia. So, I think the current PR is erroneous. 

> you can also add a DuelingNet similar to ImplicitQuantileNet into dqn.jl

Ok, looking into it.
